app/src/pages/playground/playgroundUtils.ts

```diff
@@ -123,6 +128,51 @@ function getOutputFromAttributes(parsedAttributes: unknown) {
   };
 }
 
+/**
+ * Attempts to infer the provider of the model from the model name.
+ * @param modelName the model name to get the provider from
+ * @returns the provider of the model @default AZURE_OPENAI
+ *
+ * NB: Only exported for testing
+ */
+export function getModelProviderFromModelName(
+  modelName: string
+): ModelProvider {
+  if (modelName.includes("gpt") || modelName.includes("o1")) {
+    return "OPENAI";
+  }
+  if (modelName.includes("claude")) {
+    return "ANTHROPIC";
+  }
+  // Azure OpenAI deployments have underlying OpenAI models with known names, however, when calling the llm
+  // users must use the deployment name which is a user-defined string.
+  // For now if the model name does not match any known models, we will default to Azure OpenAI.
+  return "AZURE_OPENAI";
```

Comment:
this can be changed, openai fallback makes more sense, but this might actually catch some cases where the dpeloyment name is picked in instrumentation up and this correctly falls to azure_openai - also not sure what actually gets picked up in instrumentation, the model or the deployment name

Reply:
Yeah this feels maybe dangerous. we might just want to fall back to a default if the provider cannot be inferred